state-chain/custom-rpc/src/lib.rs

```diff
@@ -1121,6 +1141,28 @@ where
 		)
 	}
 
+	fn cf_subscribe_scheduled_swaps(
+		&self,
+		sink: SubscriptionSink,
+		base_asset: Asset,
+		quote_asset: Asset,
+	) -> Result<(), SubscriptionEmptyError> {
```

Comment:
I've realised there is something else we need to deal with, swaps can fail will limited liquidity, so the simulation of the first leg can final. I think what we should do is if the simulation fails, don't output the second leg as scheduled, but do output the first leg as being scheduled.

Reply:
In this case if the liquidity gets added for the first leg, since the LPs can see that it requires liquidity, then it'll be swapped, but then the no one has accounted for the second leg? that seems bad? Would it not be better to use some kind approximate price for the first leg, to the estimate the second leg input - and perhaps making it clear through the API that this second amount is an estimate or something